src/Provider/PaginationServiceProvider.php

```diff
@@ -8,102 +9,103 @@
 use Knp\Component\Pager\Event\Subscriber\Paginate\PaginationSubscriber;
 use Knp\Component\Pager\Event\Subscriber\Sortable\SortableSubscriber;
 use Knp\Component\Pager\Paginator;
+use Pimple\Container;
+use Pimple\ServiceProviderInterface;
 use Silex\Application;
+use Silex\Api\BootableProviderInterface;
 use Silex\Provider\TranslationServiceProvider;
 use Silex\Provider\TwigServiceProvider;
-use Silex\Provider\UrlGeneratorServiceProvider;
-use Silex\ServiceProviderInterface;
+use Silex\Provider\LocaleServiceProvider;
 use Symfony\Bundle\FrameworkBundle\Templating\Helper\RouterHelper;
 use Symfony\Component\HttpKernel\KernelEvents;
 
-class PaginationServiceProvider implements ServiceProviderInterface
+class PaginationServiceProvider implements ServiceProviderInterface, BootableProviderInterface
 {
     /**
      * {@inheritdoc}
      */
-    public function register(Application $app)
+    public function register(Container $app)
     {
         if (!isset($app['twig'])) {
-            $app->register(new TwigServiceProvider());
-        }
-        if (!isset($app['url_generator'])) {
-            $app->register(new UrlGeneratorServiceProvider());
+            throw new \LogicException('You must register the TwigServiceProvider to use the PaginationServiceProvider.');
         }
         if (!isset($app['translator'])) {
-            $app->register(new TranslationServiceProvider());
+            throw new \LogicException('You must register the TranslationServiceProvider to use the PaginationServiceProvider.');
```

Comment:
also [`RoutingServiceProvider`](https://github.com/silexphp/Silex/blob/master/src/Silex/Provider/RoutingServiceProvider.php#L48) for `$app['url_generator']`